File: converted_source/umover.lfm.java

import java.awt.*;
import java.awt.event.*;
import javax.swing.*;
import javax.swing.event.*;
import javax.swing.table.*;
// import common.*;

public class MoverWindow extends JFrame {
    private static final long serialVersionUID = 1L;
    ImagePanel previewImage;
    ImagePanel leftPlantFileChangedIndicator;
    ImagePanel rightPlantFileChangedIndicator;
    SpeedButton helpButton;
    JTextField leftPlantFileNameEdit;
    JButton leftOpenClose;
    JTextField rightPlantFileNameEdit;
    JButton rightOpenClose;
    JButton newFile;
    JButton transfer;
    JButton close;
    JButton undo;
    JButton redo;
    JButton cut;
    JButton copy;
    JButton paste;
    JButton rename;
    JButton duplicate;
    JButton delete;
    JTable leftPlantList;
    JTable rightPlantList;
    public JPanel mainContentPane;
    public JPanel delphiPanel;
    
    public MoverWindow() {
        super();
        this.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE); // HIDE_ON_CLOSE DO_NOTHING_ON_CLOSE
        initialize();
    }
    
    private void initialize() {
        this.setSize(new Dimension(705, 325));
        this.setContentPane(getMainContentPane());
        this.setTitle("Plant Mover");
    }
    
    private JPanel getMainContentPane() {
        if (mainContentPane == null) {
            mainContentPane = new JPanel();
            mainContentPane.setLayout(new BorderLayout());
            mainContentPane.add(getDelphiPanel(), BorderLayout.CENTER);
        }
        return mainContentPane;
    }
    public void OnClose(ActionEvent event) {
        System.out.println("Closed");
    }
    
    public JPanel getDelphiPanel() {
        if (delphiPanel != null) return delphiPanel;
        Toolkit toolkit = java.awt.Toolkit.getDefaultToolkit();
        delphiPanel = new JPanel();
        delphiPanel.setLayout(null);
        // -- delphiPanel.setLayout(BoxLayout(contentPane, BoxLayout.Y_AXIS));
        // Font buttonFont = new Font("Arial Narrow", Font.PLAIN, 9);
        // Insets buttonMargin = new Insets(1, 1, 1, 1);
        this.setBounds(439, 193, 400, 373  + 80); // extra for title bar and menu
        //  --------------- UNHANDLED ATTRIBUTE: this.TextHeight = 14;
        //  --------------- UNHANDLED ATTRIBUTE: this.Scaled = False;
        //  --------------- UNHANDLED ATTRIBUTE: this.KeyPreview = True;
        //  --------------- UNHANDLED ATTRIBUTE: this.OnResize = FormResize;
        this.addKeyListener(new java.awt.event.KeyAdapter() {
            public void keyPressed(java.awt.event.KeyEvent event) {
                FormKeyDown(event);
            }
        });
        //  --------------- UNHANDLED ATTRIBUTE: this.PixelsPerInch = 96;
        //  --------------- UNHANDLED ATTRIBUTE: this.AutoScroll = False;
        //  --------------- UNHANDLED ATTRIBUTE: this.Position = poScreenCenter;
        //  --------------- UNHANDLED ATTRIBUTE: this.OnCreate = FormCreate;
        //  --------------- UNHANDLED ATTRIBUTE: this.OnClose = FormClose;
        
        
        this.previewImage = new ImagePanel(); // No image was set
        this.previewImage.setBounds(150, 4, 100, 71);
        
        Image leftPlantFileChangedIndicatorImage = toolkit.createImage("../resources/MoverForm_leftPlantFileChangedIndicator.png");
        this.leftPlantFileChangedIndicator = new ImagePanel(new ImageIcon(leftPlantFileChangedIndicatorImage));
        this.leftPlantFileChangedIndicator.setBounds(4, 30, 16, 16);
        
        Image rightPlantFileChangedIndicatorImage = toolkit.createImage("../resources/MoverForm_rightPlantFileChangedIndicator.png");
        this.rightPlantFileChangedIndicator = new ImagePanel(new ImageIcon(rightPlantFileChangedIndicatorImage));
        this.rightPlantFileChangedIndicator.setBounds(262, 30, 16, 16);
        
        Image helpButtonImage = toolkit.createImage("../resources/MoverForm_helpButton.png");
        this.helpButton = new SpeedButton("Help", new ImageIcon(helpButtonImage));
        this.helpButton.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                helpButtonClick(event);
            }
        });
        this.helpButton.setMnemonic(KeyEvent.VK_H);
        this.helpButton.setBounds(152, 321, 100, 24);
        
        this.leftPlantFileNameEdit = new JTextField("c:\PlantStudio\plants1.pla");
        this.leftPlantFileNameEdit.setEditable(false);
        this.leftPlantFileNameEdit.setBounds(2, 4, 140, 20);
        //  --------------- UNHANDLED ATTRIBUTE: this.leftPlantFileNameEdit.AutoSelect = False;
        
        this.leftOpenClose = new JButton("Open...");
        this.leftOpenClose.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                leftOpenCloseClick(event);
            }
        });
        this.leftOpenClose.setMnemonic(KeyEvent.VK_O);
        this.leftOpenClose.setBounds(24, 28, 50, 21);
        
        this.rightPlantFileNameEdit = new JTextField("c:\PlantStudio\newplants.pla");
        this.rightPlantFileNameEdit.setEditable(false);
        this.rightPlantFileNameEdit.setBounds(258, 4, 140, 20);
        //  --------------- UNHANDLED ATTRIBUTE: this.rightPlantFileNameEdit.AutoSelect = False;
        
        this.rightOpenClose = new JButton("Open...");
        this.rightOpenClose.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                rightOpenCloseClick(event);
            }
        });
        this.rightOpenClose.setMnemonic(KeyEvent.VK_O);
        this.rightOpenClose.setBounds(284, 27, 50, 21);
        
        this.newFile = new JButton("New...");
        this.newFile.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                newFileClick(event);
            }
        });
        this.newFile.setMnemonic(KeyEvent.VK_N);
        this.newFile.setBounds(337, 27, 50, 21);
        
        this.transfer = new JButton("Transfer >>");
        this.transfer.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                transferClick(event);
            }
        });
        this.transfer.setMnemonic(KeyEvent.VK_T);
        this.transfer.setBounds(151, 108, 100, 21);
        
        this.close = new JButton("Quit");
        this.close.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                closeClick(event);
            }
        });
        this.close.setMnemonic(KeyEvent.VK_Q);
        this.close.setBounds(151, 349, 100, 21);
        
        this.undo = new JButton("Undo duplicate");
        this.undo.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                undoClick(event);
            }
        });
        this.undo.setMnemonic(KeyEvent.VK_U);
        this.undo.setBounds(151, 273, 100, 21);
        
        this.redo = new JButton("Redo last");
        this.redo.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                redoClick(event);
            }
        });
        this.redo.setMnemonic(KeyEvent.VK_R);
        this.redo.setBounds(151, 297, 100, 21);
        
        this.cut = new JButton("Cut (Ctrl-X)");
        this.cut.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                cutClick(event);
            }
        });
        this.cut.setMnemonic(KeyEvent.VK_C);
        this.cut.setBounds(151, 132, 100, 21);
        
        this.copy = new JButton("Copy (Ctrl-C)");
        this.copy.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                copyClick(event);
            }
        });
        this.copy.setMnemonic(KeyEvent.VK_O);
        this.copy.setBounds(151, 155, 100, 21);
        
        this.paste = new JButton("Paste (Ctrl-V)");
        this.paste.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                pasteClick(event);
            }
        });
        this.paste.setMnemonic(KeyEvent.VK_P);
        this.paste.setBounds(151, 179, 100, 21);
        
        this.rename = new JButton("Rename...");
        this.rename.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                renameClick(event);
            }
        });
        this.rename.setMnemonic(KeyEvent.VK_A);
        this.rename.setBounds(151, 250, 100, 21);
        
        this.duplicate = new JButton("Duplicate");
        this.duplicate.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                duplicateClick(event);
            }
        });
        this.duplicate.setMnemonic(KeyEvent.VK_D);
        this.duplicate.setBounds(151, 202, 100, 21);
        
        this.delete = new JButton("Delete");
        this.delete.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                deleteClick(event);
            }
        });
        this.delete.setMnemonic(KeyEvent.VK_E);
        this.delete.setBounds(151, 226, 100, 21);
        
        this.leftPlantList = new JTable(new DefaultTableModel());
        this.leftPlantList.setBounds(4, 52, 140, 315);
        //  --------------- UNHANDLED ATTRIBUTE: this.leftPlantList.OnEndDrag = leftPlantListEndDrag;
        //  --------------- UNHANDLED ATTRIBUTE: this.leftPlantList.OnDrawCell = leftPlantListDrawCell;
        //  --------------- UNHANDLED ATTRIBUTE: this.leftPlantList.OnDblClick = leftPlantListDblClick;
        //  --------------- UNHANDLED ATTRIBUTE: this.leftPlantList.OnDragOver = leftPlantListDragOver;
        //  --------------- UNHANDLED ATTRIBUTE: this.leftPlantList.DefaultDrawing = False;
        //  --------------- UNHANDLED ATTRIBUTE: this.leftPlantList.FixedRows = 0;
        this.leftPlantList.addKeyListener(new java.awt.event.KeyAdapter() {
            public void keyPressed(java.awt.event.KeyEvent event) {
                leftPlantListKeyDown(event);
            }
        });
        this.leftPlantList.addMouseListener(new java.awt.event.MouseAdapter() {
            public void mousePressed(java.awt.event.MouseEvent event) {
                leftPlantListMouseDown(event);
            }
        });
        //  --------------- UNHANDLED ATTRIBUTE: this.leftPlantList.ColCount = 1;
        //  --------------- UNHANDLED ATTRIBUTE: this.leftPlantList.DefaultRowHeight = 16;
        //  --------------- UNHANDLED ATTRIBUTE: this.leftPlantList.ScrollBars = ssVertical;
        //  --------------- UNHANDLED ATTRIBUTE: this.leftPlantList.FixedCols = 0;
        //  --------------- UNHANDLED ATTRIBUTE: this.leftPlantList.GridLineWidth = 0;
        //  --------------- UNHANDLED ATTRIBUTE: this.leftPlantList.Options = [];
        
        this.rightPlantList = new JTable(new DefaultTableModel());
        this.rightPlantList.setBounds(258, 52, 140, 317);
        //  --------------- UNHANDLED ATTRIBUTE: this.rightPlantList.OnEndDrag = rightPlantListEndDrag;
        //  --------------- UNHANDLED ATTRIBUTE: this.rightPlantList.OnDrawCell = rightPlantListDrawCell;
        //  --------------- UNHANDLED ATTRIBUTE: this.rightPlantList.OnDblClick = rightPlantListDblClick;
        //  --------------- UNHANDLED ATTRIBUTE: this.rightPlantList.OnDragOver = rightPlantListDragOver;
        //  --------------- UNHANDLED ATTRIBUTE: this.rightPlantList.DefaultDrawing = False;
        //  --------------- UNHANDLED ATTRIBUTE: this.rightPlantList.FixedRows = 0;
        this.rightPlantList.addKeyListener(new java.awt.event.KeyAdapter() {
            public void keyPressed(java.awt.event.KeyEvent event) {
                rightPlantListKeyDown(event);
            }
        });
        //  --------------- UNHANDLED ATTRIBUTE: this.rightPlantList.ColCount = 1;
        //  --------------- UNHANDLED ATTRIBUTE: this.rightPlantList.DefaultRowHeight = 16;
        //  --------------- UNHANDLED ATTRIBUTE: this.rightPlantList.FixedCols = 0;
        //  --------------- UNHANDLED ATTRIBUTE: this.rightPlantList.GridLineWidth = 0;
        this.rightPlantList.addMouseListener(new java.awt.event.MouseAdapter() {
            public void mousePressed(java.awt.event.MouseEvent event) {
                rightPlantListMouseDown(event);
            }
        });
        
        delphiPanel.add(previewImage);
        delphiPanel.add(leftPlantFileChangedIndicator);
        delphiPanel.add(rightPlantFileChangedIndicator);
        delphiPanel.add(helpButton);
        delphiPanel.add(leftPlantFileNameEdit);
        delphiPanel.add(leftOpenClose);
        delphiPanel.add(rightPlantFileNameEdit);
        delphiPanel.add(rightOpenClose);
        delphiPanel.add(newFile);
        delphiPanel.add(transfer);
        delphiPanel.add(close);
        delphiPanel.add(undo);
        delphiPanel.add(redo);
        delphiPanel.add(cut);
        delphiPanel.add(copy);
        delphiPanel.add(paste);
        delphiPanel.add(rename);
        delphiPanel.add(duplicate);
        delphiPanel.add(delete);
        delphiPanel.add(leftPlantList);
        delphiPanel.add(rightPlantList);
        return delphiPanel;
    }
    
        
    public void FormKeyDown(KeyEvent event) {
        System.out.println("FormKeyDown");
    }
        
    public void closeClick(ActionEvent event) {
        System.out.println("closeClick");
    }
        
    public void copyClick(ActionEvent event) {
        System.out.println("copyClick");
    }
        
    public void cutClick(ActionEvent event) {
        System.out.println("cutClick");
    }
        
    public void deleteClick(ActionEvent event) {
        System.out.println("deleteClick");
    }
        
    public void duplicateClick(ActionEvent event) {
        System.out.println("duplicateClick");
    }
        
    public void helpButtonClick(ActionEvent event) {
        System.out.println("helpButtonClick");
    }
        
    public void leftOpenCloseClick(ActionEvent event) {
        System.out.println("leftOpenCloseClick");
    }
        
    public void leftPlantListKeyDown(KeyEvent event) {
        System.out.println("leftPlantListKeyDown");
    }
        
    public void leftPlantListMouseDown(MouseEvent event) {
        System.out.println("leftPlantListMouseDown");
    }
        
    public void newFileClick(ActionEvent event) {
        System.out.println("newFileClick");
    }
        
    public void pasteClick(ActionEvent event) {
        System.out.println("pasteClick");
    }
        
    public void redoClick(ActionEvent event) {
        System.out.println("redoClick");
    }
        
    public void renameClick(ActionEvent event) {
        System.out.println("renameClick");
    }
        
    public void rightOpenCloseClick(ActionEvent event) {
        System.out.println("rightOpenCloseClick");
    }
        
    public void rightPlantListKeyDown(KeyEvent event) {
        System.out.println("rightPlantListKeyDown");
    }
        
    public void rightPlantListMouseDown(MouseEvent event) {
        System.out.println("rightPlantListMouseDown");
    }
        
    public void transferClick(ActionEvent event) {
        System.out.println("transferClick");
    }
        
    public void undoClick(ActionEvent event) {
        System.out.println("undoClick");
    }

    public static void main(String[] args) {
        SwingUtilities.invokeLater(new Runnable() {
            public void run() {
                MoverWindow thisClass = new MoverWindow();
                thisClass.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
                thisClass.setVisible(true);
            }
        });
    }

}

File: converted_source/umover.lfm.py
from common import *

#Useage:
#        import MoverWindow
#        window = MoverWindow.MoverWindow()
#        window.visible = 1

class MoverWindow(JFrame):
    def __init__(self, title='Plant Mover'):
        JFrame.__init__(self, title, windowClosing=self.OnClose)
        self.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE) # HIDE_ON_CLOSE DO_NOTHING_ON_CLOSE
        self.BuildWindow()
        
    def OnClose(self, event):
        print "Closed"
        
    # a menu is top item on the menu bar and can be pulled down for a list of menu items
    def createMenu(self, name, mnemonic=None, menuBar = None):
        menu = JMenu(name)
        if mnemonic:
            menu.setMnemonic(mnemonic)
        if menuBar:
            menuBar.add(menu)
        return menu
        
    def BuildWindow(self):
        toolkit = java.awt.Toolkit.getDefaultToolkit()
        contentPane = self.getContentPane()
        contentPane.setLayout(None)
        # -- contentPane.setLayout(BoxLayout(contentPane, BoxLayout.Y_AXIS))
        buttonFont = Font("Arial Narrow", Font.PLAIN, 9)
        buttonMargin = Insets(1, 1, 1, 1)
        self.setBounds(439, 193, 400, 373  + 80) # extra for title bar and menu
        #  --------------- UNHANDLED ATTRIBUTE: self.TextHeight = 14
        #  --------------- UNHANDLED ATTRIBUTE: self.Scaled = False
        #  --------------- UNHANDLED ATTRIBUTE: self.KeyPreview = True
        #  --------------- UNHANDLED ATTRIBUTE: self.OnResize = FormResize
        self.keyPressed = self.FormKeyDown
        #  --------------- UNHANDLED ATTRIBUTE: self.PixelsPerInch = 96
        #  --------------- UNHANDLED ATTRIBUTE: self.AutoScroll = False
        #  --------------- UNHANDLED ATTRIBUTE: self.Position = poScreenCenter
        #  --------------- UNHANDLED ATTRIBUTE: self.OnCreate = FormCreate
        #  --------------- UNHANDLED ATTRIBUTE: self.OnClose = FormClose
        
        self.previewImage = ImagePanel() # No image was set
        self.previewImage.setBounds(150, 4, 100, 71)
        self.leftPlantFileChangedIndicatorImage = toolkit.createImage("../resources/MoverForm_leftPlantFileChangedIndicator.png")
        self.leftPlantFileChangedIndicator = ImagePanel(ImageIcon(self.leftPlantFileChangedIndicatorImage))
        self.leftPlantFileChangedIndicator.setBounds(4, 30, 16, 16)
        self.rightPlantFileChangedIndicatorImage = toolkit.createImage("../resources/MoverForm_rightPlantFileChangedIndicator.png")
        self.rightPlantFileChangedIndicator = ImagePanel(ImageIcon(self.rightPlantFileChangedIndicatorImage))
        self.rightPlantFileChangedIndicator.setBounds(262, 30, 16, 16)
        self.helpButtonImage = toolkit.createImage("../resources/MoverForm_helpButton.png")
        self.helpButton = SpeedButton('Help', ImageIcon(self.helpButtonImage), actionPerformed=self.helpButtonClick, font=buttonFont, margin=buttonMargin)
        self.helpButton.setMnemonic(KeyEvent.VK_H)
        self.helpButton.setBounds(152, 321, 100, 24)
        self.leftPlantFileNameEdit = JTextField('c:\PlantStudio\plants1.pla', 10, editable=0)
        self.leftPlantFileNameEdit.setBounds(2, 4, 140, 20)
        #  --------------- UNHANDLED ATTRIBUTE: self.leftPlantFileNameEdit.AutoSelect = False
        self.leftOpenClose = JButton('Open...', actionPerformed=self.leftOpenCloseClick, font=buttonFont, margin=buttonMargin)
        self.leftOpenClose.setMnemonic(KeyEvent.VK_O)
        self.leftOpenClose.setBounds(24, 28, 50, 21)
        self.rightPlantFileNameEdit = JTextField('c:\PlantStudio\newplants.pla', 10, editable=0)
        self.rightPlantFileNameEdit.setBounds(258, 4, 140, 20)
        #  --------------- UNHANDLED ATTRIBUTE: self.rightPlantFileNameEdit.AutoSelect = False
        self.rightOpenClose = JButton('Open...', actionPerformed=self.rightOpenCloseClick, font=buttonFont, margin=buttonMargin)
        self.rightOpenClose.setMnemonic(KeyEvent.VK_O)
        self.rightOpenClose.setBounds(284, 27, 50, 21)
        self.newFile = JButton('New...', actionPerformed=self.newFileClick, font=buttonFont, margin=buttonMargin)
        self.newFile.setMnemonic(KeyEvent.VK_N)
        self.newFile.setBounds(337, 27, 50, 21)
        self.transfer = JButton('Transfer >>', actionPerformed=self.transferClick, font=buttonFont, margin=buttonMargin)
        self.transfer.setMnemonic(KeyEvent.VK_T)
        self.transfer.setBounds(151, 108, 100, 21)
        self.close = JButton('Quit', actionPerformed=self.closeClick, font=buttonFont, margin=buttonMargin)
        self.close.setMnemonic(KeyEvent.VK_Q)
        self.close.setBounds(151, 349, 100, 21)
        self.undo = JButton('Undo duplicate', actionPerformed=self.undoClick, font=buttonFont, margin=buttonMargin)
        self.undo.setMnemonic(KeyEvent.VK_U)
        self.undo.setBounds(151, 273, 100, 21)
        self.redo = JButton('Redo last', actionPerformed=self.redoClick, font=buttonFont, margin=buttonMargin)
        self.redo.setMnemonic(KeyEvent.VK_R)
        self.redo.setBounds(151, 297, 100, 21)
        self.cut = JButton('Cut (Ctrl-X)', actionPerformed=self.cutClick, font=buttonFont, margin=buttonMargin)
        self.cut.setMnemonic(KeyEvent.VK_C)
        self.cut.setBounds(151, 132, 100, 21)
        self.copy = JButton('Copy (Ctrl-C)', actionPerformed=self.copyClick, font=buttonFont, margin=buttonMargin)
        self.copy.setMnemonic(KeyEvent.VK_O)
        self.copy.setBounds(151, 155, 100, 21)
        self.paste = JButton('Paste (Ctrl-V)', actionPerformed=self.pasteClick, font=buttonFont, margin=buttonMargin)
        self.paste.setMnemonic(KeyEvent.VK_P)
        self.paste.setBounds(151, 179, 100, 21)
        self.rename = JButton('Rename...', actionPerformed=self.renameClick, font=buttonFont, margin=buttonMargin)
        self.rename.setMnemonic(KeyEvent.VK_A)
        self.rename.setBounds(151, 250, 100, 21)
        self.duplicate = JButton('Duplicate', actionPerformed=self.duplicateClick, font=buttonFont, margin=buttonMargin)
        self.duplicate.setMnemonic(KeyEvent.VK_D)
        self.duplicate.setBounds(151, 202, 100, 21)
        self.delete = JButton('Delete', actionPerformed=self.deleteClick, font=buttonFont, margin=buttonMargin)
        self.delete.setMnemonic(KeyEvent.VK_E)
        self.delete.setBounds(151, 226, 100, 21)
        self.leftPlantList = JTable(DefaultTableModel())
        self.leftPlantList.setBounds(4, 52, 140, 315)
        #  --------------- UNHANDLED ATTRIBUTE: self.leftPlantList.OnEndDrag = leftPlantListEndDrag
        #  --------------- UNHANDLED ATTRIBUTE: self.leftPlantList.OnDrawCell = leftPlantListDrawCell
        #  --------------- UNHANDLED ATTRIBUTE: self.leftPlantList.OnDblClick = leftPlantListDblClick
        #  --------------- UNHANDLED ATTRIBUTE: self.leftPlantList.OnDragOver = leftPlantListDragOver
        #  --------------- UNHANDLED ATTRIBUTE: self.leftPlantList.DefaultDrawing = False
        #  --------------- UNHANDLED ATTRIBUTE: self.leftPlantList.FixedRows = 0
        self.leftPlantList.keyPressed = self.leftPlantListKeyDown
        self.leftPlantList.mousePressed = self.leftPlantListMouseDown
        #  --------------- UNHANDLED ATTRIBUTE: self.leftPlantList.ColCount = 1
        #  --------------- UNHANDLED ATTRIBUTE: self.leftPlantList.DefaultRowHeight = 16
        #  --------------- UNHANDLED ATTRIBUTE: self.leftPlantList.ScrollBars = ssVertical
        #  --------------- UNHANDLED ATTRIBUTE: self.leftPlantList.FixedCols = 0
        #  --------------- UNHANDLED ATTRIBUTE: self.leftPlantList.GridLineWidth = 0
        #  --------------- UNHANDLED ATTRIBUTE: self.leftPlantList.Options = []
        self.rightPlantList = JTable(DefaultTableModel())
        self.rightPlantList.setBounds(258, 52, 140, 317)
        #  --------------- UNHANDLED ATTRIBUTE: self.rightPlantList.OnEndDrag = rightPlantListEndDrag
        #  --------------- UNHANDLED ATTRIBUTE: self.rightPlantList.OnDrawCell = rightPlantListDrawCell
        #  --------------- UNHANDLED ATTRIBUTE: self.rightPlantList.OnDblClick = rightPlantListDblClick
        #  --------------- UNHANDLED ATTRIBUTE: self.rightPlantList.OnDragOver = rightPlantListDragOver
        #  --------------- UNHANDLED ATTRIBUTE: self.rightPlantList.DefaultDrawing = False
        #  --------------- UNHANDLED ATTRIBUTE: self.rightPlantList.FixedRows = 0
        self.rightPlantList.keyPressed = self.rightPlantListKeyDown
        #  --------------- UNHANDLED ATTRIBUTE: self.rightPlantList.ColCount = 1
        #  --------------- UNHANDLED ATTRIBUTE: self.rightPlantList.DefaultRowHeight = 16
        #  --------------- UNHANDLED ATTRIBUTE: self.rightPlantList.FixedCols = 0
        #  --------------- UNHANDLED ATTRIBUTE: self.rightPlantList.GridLineWidth = 0
        self.rightPlantList.mousePressed = self.rightPlantListMouseDown
        contentPane.add(self.previewImage)
        contentPane.add(self.leftPlantFileChangedIndicator)
        contentPane.add(self.rightPlantFileChangedIndicator)
        contentPane.add(self.helpButton)
        contentPane.add(self.leftPlantFileNameEdit)
        contentPane.add(self.leftOpenClose)
        contentPane.add(self.rightPlantFileNameEdit)
        contentPane.add(self.rightOpenClose)
        contentPane.add(self.newFile)
        contentPane.add(self.transfer)
        contentPane.add(self.close)
        contentPane.add(self.undo)
        contentPane.add(self.redo)
        contentPane.add(self.cut)
        contentPane.add(self.copy)
        contentPane.add(self.paste)
        contentPane.add(self.rename)
        contentPane.add(self.duplicate)
        contentPane.add(self.delete)
        contentPane.add(self.leftPlantList)
        contentPane.add(self.rightPlantList)
        
    def FormKeyDown(self, event):
        print "FormKeyDown"
        
    def closeClick(self, event):
        print "closeClick"
        
    def copyClick(self, event):
        print "copyClick"
        
    def cutClick(self, event):
        print "cutClick"
        
    def deleteClick(self, event):
        print "deleteClick"
        
    def duplicateClick(self, event):
        print "duplicateClick"
        
    def helpButtonClick(self, event):
        print "helpButtonClick"
        
    def leftOpenCloseClick(self, event):
        print "leftOpenCloseClick"
        
    def leftPlantListKeyDown(self, event):
        print "leftPlantListKeyDown"
        
    def leftPlantListMouseDown(self, event):
        print "leftPlantListMouseDown"
        
    def newFileClick(self, event):
        print "newFileClick"
        
    def pasteClick(self, event):
        print "pasteClick"
        
    def redoClick(self, event):
        print "redoClick"
        
    def renameClick(self, event):
        print "renameClick"
        
    def rightOpenCloseClick(self, event):
        print "rightOpenCloseClick"
        
    def rightPlantListKeyDown(self, event):
        print "rightPlantListKeyDown"
        
    def rightPlantListMouseDown(self, event):
        print "rightPlantListMouseDown"
        
    def transferClick(self, event):
        print "transferClick"
        
    def undoClick(self, event):
        print "undoClick"
        
if __name__ == "__main__":
    window = MoverWindow()
    window.visible = 1
    window.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE)
